src/vehicle_autodrive.cpp

```diff
@@ -51,7 +53,16 @@ static constexpr int VMIPH_PER_TPS = static_cast<int>( vehicles::vmiph_per_tile
  * a multiple of TURNING_INCREMENT degrees, measured from the x axis.
  * Range: [0, NUM_ORIENTATIONS)
  */
-using orientation = int16_t;
+enum class orientation : int8_t {
```

Comment:
Mainly because units::angle is backed by a double (8 bytes) while orientation is backed by int8_t (1 byte).
orientation also has some other convenient overloads that angle lacks.

There's no x and y in orientation, so I'm confused by that remark. If you mean node_address, the reason is similar to the above. My x and y are backed by 2-byte integers instead of 4-byte integers.